api/k8s/v1/k8s.proto

```diff
@@ -419,6 +422,31 @@ message Deployment {
 
   map<string, string> labels = 4;
   map<string, string> annotations = 5;
+
+  message DeploymentStatus {
+    uint32 replicas = 1;
+    uint32 updated_replicas = 2;
+    uint32 ready_replicas = 3;
+    uint32 available_replicas = 4;
+    uint32 unavailable_replicas = 5;
+    message DeploymentCondition {
+      enum DeploymentConditionType {
+        UNSPECIFIED = 0;
+        UNKNOWN = 1;
+        AVAILABLE = 2;
+        PROGRESSING = 3;
+        REPLICA_FAILURE = 4;
+      }
+
+      DeploymentConditionType deployment_condition_type = 1;
```

Comment:
Could you add a comment here saying this is mirroring the upstream values here: https://pkg.go.dev/k8s.io/api/apps/v1#DeploymentConditionType